File: experiments-local/experiment-15/runs/agent=omp_language=go_model=openrouter/deepseek/deepseek-v3.2_tooling=none/rep2/models/book.go
package models

import (
	"time"
)

type Book struct {
	ID        string    `json:"id"`
	Title     string    `json:"title"`
	Author    string    `json:"author"`
	Year      int       `json:"year"`
	ISBN      string    `json:"isbn"`
	CreatedAt time.Time `json:"created_at"`
	UpdatedAt time.Time `json:"updated_at"`
}

type BookRequest struct {
	Title  string `json:"title" validate:"required"`
	Author string `json:"author" validate:"required"`
	Year   int    `json:"year" validate:"min=0,max=2100"`
	ISBN   string `json:"isbn"`
}

type BookResponse struct {
	ID        string    `json:"id"`
	Title     string    `json:"title"`
	Author    string    `json:"author"`
	Year      int       `json:"year"`
	ISBN      string    `json:"isbn"`
	CreatedAt time.Time `json:"created_at"`
	UpdatedAt time.Time `json:"updated_at"`
}

type BookStore interface {
	Create(book *BookRequest) (*Book, error)
	GetByID(id string) (*Book, error)
	List(authorFilter string) ([]Book, error)
	Update(id string, book *BookRequest) (*Book, error)
	Delete(id string) error
}

File: experiments-local/experiment-15/runs/agent=omp_language=typescript_model=openrouter/deepseek/deepseek-v3.2_tooling=none/rep2/src/entities/Book.ts
import { Entity, PrimaryGeneratedColumn, Column, CreateDateColumn, UpdateDateColumn } from 'typeorm';
import { IsNotEmpty, IsString, IsInt, Min, Max, IsOptional, IsISBN } from 'class-validator';

@Entity()
export class Book {
  @PrimaryGeneratedColumn('uuid')
  id!: string;

  @Column()
  @IsNotEmpty({ message: 'Title is required' })
  @IsString()
  title!: string;

  @Column()
  @IsNotEmpty({ message: 'Author is required' })
  @IsString()
  author!: string;

  @Column('integer')
  @IsInt()
  @Min(1000, { message: 'Year must be at least 1000' })
  @Max(new Date().getFullYear(), { message: 'Year cannot be in the future' })
  year!: number;

  @Column({ unique: true })
  @IsNotEmpty({ message: 'ISBN is required' })
  @IsISBN()
  isbn!: string;

  @CreateDateColumn()
  createdAt!: Date;

  @UpdateDateColumn()
  updatedAt!: Date;
}

export class CreateBookDto {
  @IsNotEmpty({ message: 'Title is required' })
  @IsString()
  title!: string;

  @IsNotEmpty({ message: 'Author is required' })
  @IsString()
  author!: string;

  @IsInt()
  @Min(1000, { message: 'Year must be at least 1000' })
  @Max(new Date().getFullYear(), { message: 'Year cannot be in the future' })
  year!: number;

  @IsNotEmpty({ message: 'ISBN is required' })
  @IsISBN()
  isbn!: string;
}

export class UpdateBookDto {
  @IsOptional()
  @IsString()
  title?: string;

  @IsOptional()
  @IsString()
  author?: string;

  @IsOptional()
  @IsInt()
  @Min(1000, { message: 'Year must be at least 1000' })
  @Max(new Date().getFullYear(), { message: 'Year cannot be in the future' })
  year?: number;

  @IsOptional()
  @IsISBN()
  isbn?: string;
}